### ISSUE TITLE
when searching,i had this error

### ISSUE THREAD

**Issue by badaozhenjun** [tillson/git-hound#27]
git-hound version
```
latest realease so far
```
go version:
```
go version go1.13.4 linux/amd64
```
cmd:
```
echo jd.com | ./git-hound --dig-files --dig-commits --many-results --regex-file halcorp-api-keys.txt --results-only | tee api_tokens.txt
```
regex-file content:
```
(((https?|ftp):\/\/)*([a-zA-z0-9][-a-zA-z0-9]{0,62}\.)*jd\.com(:\d+)*([A-Za-z0-9\-\._~:\/\?#\[\]@!$&'\(\)\*\+,;\=]*)?)
```
start error after print some correct string
```
encoding/asn1.parseField(0xafba20, 0xc017d9eed0, 0x199, 0xc007eee400, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/opt/go/libexec/src/encoding/asn1/asn1.go:901 +0x1bb4
encoding/asn1.UnmarshalWithParams(0xc007eee400, 0x2, 0x2, 0xa6b820, 0xc017d9eed0, 0x0, 0x0, 0x10, 0x0, 0x0, ...)
	/usr/local/opt/go/libexec/src/encoding/asn1/asn1.go:1061 +0x1af
encoding/asn1.Unmarshal(...)
	/usr/local/opt/go/libexec/src/encoding/asn1/asn1.go:1054
crypto/x509.parseCertificate(0xc002e72300, 0x6d9, 0xbf6, 0xa6b860)
	/usr/local/opt/go/libexec/src/crypto/x509/x509.go:1402 +0xc03
crypto/x509.ParseCertificate(0xc00810b00a, 0x6d9, 0xbf6, 0xc00bd9bfc0, 0xb4936b7b3e125ec3, 0xd6990624d192e819)
	/usr/local/opt/go/libexec/src/crypto/x509/x509.go:1569 +0x11d
crypto/tls.(*Conn).verifyServerCertificate(0xc000042a80, 0xc013b5b650, 0x2, 0x2, 0xb9b, 0x0)
	/usr/local/opt/go/libexec/src/crypto/tls/handshake_client.go:815 +0x61f
crypto/tls.(*clientHandshakeState).doFullHandshake(0xc015753dc8, 0xc0002455e0, 0x6c)
	/usr/local/opt/go/libexec/src/crypto/tls/handshake_client.go:454 +0x1869
crypto/tls.(*clientHandshakeState).handshake(0xc015753dc8, 0xc0014e4900, 0x0)
	/usr/local/opt/go/libexec/src/crypto/tls/handshake_client.go:399 +0x3fb
crypto/tls.(*Conn).clientHandshake(0xc000042a80, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/crypto/tls/handshake_client.go:208 +0x2cd
crypto/tls.(*Conn).Handshake(0xc000042a80, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/crypto/tls/conn.go:1343 +0xef
net/http.(*persistConn).addTLS.func2(0x0, 0xc000042a80, 0xc0196f3db0, 0xc0147aaba0)
	/usr/local/opt/go/libexec/src/net/http/transport.go:1190 +0x42
created by net/http.(*persistConn).addTLS
	/usr/local/opt/go/libexec/src/net/http/transport.go:1186 +0x1ab

goroutine 163 [chan receive, 1 minutes]:
github.com/tillson/git-hound/internal/app.Dig(0xc0006bf392, 0x1f, 0xc0006bf3e0, 0x13, 0xc000933d80, 0x3a, 0xb8c4bf, 0x4, 0xc000025bd8, 0x6, ...)
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/dig.go:39 +0xf6
github.com/tillson/git-hound/internal/app.ScanAndPrintResult(0xc000392660, 0xc0006bf392, 0x1f, 0xc0006bf3e0, 0x13, 0xc000933d80, 0x3a, 0xb8c4bf, 0x4, 0xc000025bd8, ...)
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/keyword_scan.go:64 +0x60c
created by github.com/tillson/git-hound/internal/app.SearchGitHub
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/search.go:169 +0x8ad

goroutine 502 [chan receive]:
github.com/go-git/go-git/v5/storage/filesystem/dotgit.(*PackWriter).waitBuildIndex(...)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/storage/filesystem/dotgit/writers.go:81
github.com/go-git/go-git/v5/storage/filesystem/dotgit.(*PackWriter).Close(0xc00579c000, 0x0, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/storage/filesystem/dotgit/writers.go:108 +0x97
github.com/go-git/go-git/v5/utils/ioutil.CheckClose(0x7f901eed3460, 0xc00579c000, 0xc0002bb3d8)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/utils/ioutil/common.go:89 +0x31
github.com/go-git/go-git/v5/plumbing/format/packfile.WritePackfileToObjectStorage(0x7f901eed33f8, 0xc003f8c5a0, 0xc8f260, 0xc004690aa0, 0x0, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/common.go:62 +0x163
github.com/go-git/go-git/v5/plumbing/format/packfile.UpdateObjectStorage(0xca70c0, 0xc003f8c5a0, 0xc8f260, 0xc004690aa0, 0xc003f8c5a0, 0xc8f260)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/common.go:31 +0x236
github.com/go-git/go-git/v5.(*Remote).fetchPack(0xc0003c2480, 0xc9f200, 0xc00615cea0, 0xc0002bb730, 0xc9c440, 0xc000ff1ee8, 0xc00073faa0, 0x0, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/remote.go:391 +0x1bf
github.com/go-git/go-git/v5.(*Remote).fetch(0xc0003c2480, 0xc9f200, 0xc00615cea0, 0xc0002bb730, 0x0, 0x0, 0x0, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/remote.go:328 +0x60e
github.com/go-git/go-git/v5.(*Repository).fetchAndUpdateReferences(0xc006179a40, 0xc9f200, 0xc00615cea0, 0xc0002bb730, 0xb8bfd7, 0x4, 0xc0002bb790, 0x9ee997, 0xcaa080)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/repository.go:877 +0xbf
github.com/go-git/go-git/v5.(*Repository).clone(0xc006179a40, 0xc9f200, 0xc00615cea0, 0xc005fc5340, 0x0, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/repository.go:744 +0x2a9
github.com/go-git/go-git/v5.PlainCloneContext(0xc9f200, 0xc00615cea0, 0xc00053f140, 0x31, 0x0, 0xc005fc5340, 0x31, 0xc0001f8880, 0x47656d)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/repository.go:360 +0xcc
github.com/tillson/git-hound/internal/app.digHelper(0xc001bbd47f, 0x23, 0xc001bbd4d1, 0x1a, 0xc000079860, 0x45, 0xb8c4bf, 0x4, 0xc000025bd8, 0x6, ...)
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/dig.go:54 +0x4f3
github.com/tillson/git-hound/internal/app.Dig.func1(0xc000065a40, 0xc001bbd47f, 0x23, 0xc001bbd4d1, 0x1a, 0xc000079860, 0x45, 0xb8c4bf, 0x4, 0xc000025bd8, ...)
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/dig.go:35 +0x60
created by github.com/tillson/git-hound/internal/app.Dig
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/dig.go:34 +0xca

goroutine 256 [chan receive]:
github.com/tillson/git-hound/internal/app.Dig(0xc001bbd47f, 0x23, 0xc001bbd4d1, 0x1a, 0xc000079860, 0x45, 0xb8c4bf, 0x4, 0xc000025bd8, 0x6, ...)
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/dig.go:39 +0xf6
github.com/tillson/git-hound/internal/app.ScanAndPrintResult(0xc000392660, 0xc001bbd47f, 0x23, 0xc001bbd4d1, 0x1a, 0xc000079860, 0x45, 0xb8c4bf, 0x4, 0xc000025bd8, ...)
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/keyword_scan.go:64 +0x60c
created by github.com/tillson/git-hound/internal/app.SearchGitHub
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/search.go:169 +0x8ad

goroutine 1060 [select]:
net/http.(*persistConn).readLoop(0xc00011ed80)
	/usr/local/opt/go/libexec/src/net/http/transport.go:1798 +0x95a
created by net/http.(*Transport).dialConn
	/usr/local/opt/go/libexec/src/net/http/transport.go:1357 +0xae8

goroutine 966 [runnable]:
compress/zlib.(*reader).Reset(0xc018957db0, 0xc8f1e0, 0xc001128680, 0x0, 0x0, 0x0, 0x10, 0x10)
	/usr/local/opt/go/libexec/src/compress/zlib/reader.go:173 +0x492
github.com/go-git/go-git/v5/plumbing/format/packfile.(*Scanner).copyObject(0xc017b00c60, 0xc8ecc0, 0xc017b00d20, 0x0, 0x0, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/scanner.go:329 +0x122
github.com/go-git/go-git/v5/plumbing/format/packfile.(*Scanner).NextObject(0xc017b00c60, 0xc8ecc0, 0xc017b00d20, 0x1, 0x2, 0xc00f2199e0, 0x1)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/scanner.go:314 +0x5a
github.com/go-git/go-git/v5/plumbing/format/packfile.(*Parser).indexObjects(0xc0175a7ef0, 0x0, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/parser.go:222 +0x204
github.com/go-git/go-git/v5/plumbing/format/packfile.(*Parser).Parse(0xc0175a7ef0, 0x0, 0x0, 0x0, 0xc00159b528, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/parser.go:138 +0x61
github.com/go-git/go-git/v5/storage/filesystem/dotgit.(*PackWriter).buildIndex(0xc01880e850)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/storage/filesystem/dotgit/writers.go:67 +0x1c4
created by github.com/go-git/go-git/v5/storage/filesystem/dotgit.newPackWrite
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/storage/filesystem/dotgit/writers.go:53 +0x325

goroutine 850 [runnable]:
compress/flate.(*decompressor).huffSym(0xc000100600, 0xc000100628, 0x88, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/compress/flate/inflate.go:707 +0x24a
compress/flate.(*decompressor).huffmanBlock(0xc000100600)
	/usr/local/opt/go/libexec/src/compress/flate/inflate.go:494 +0xd0
compress/flate.(*decompressor).nextBlock(0xc000100600)
	/usr/local/opt/go/libexec/src/compress/flate/inflate.go:327 +0x16d
compress/flate.(*decompressor).Read(0xc000100600, 0xc0165b6000, 0x15fe00, 0x15fe00, 0x8000, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/compress/flate/inflate.go:347 +0x77
compress/zlib.(*reader).Read(0xc00350e9b0, 0xc0165b6000, 0x15fe00, 0x15fe00, 0x8000, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/compress/zlib/reader.go:94 +0x76
bytes.(*Buffer).ReadFrom(0xc016de3200, 0x7f901eed35d8, 0xc00350e9b0, 0x7f901eed3618, 0xc016de3200, 0x465201)
	/usr/local/opt/go/libexec/src/bytes/buffer.go:207 +0xbd
io.copyBuffer(0xc8ecc0, 0xc016de3200, 0x7f901eed35d8, 0xc00350e9b0, 0xc0027ec000, 0x8000, 0x8000, 0x0, 0x0, 0x30000c0038739c0)
	/usr/local/opt/go/libexec/src/io/io.go:388 +0x2fc
io.CopyBuffer(...)
	/usr/local/opt/go/libexec/src/io/io.go:375
github.com/go-git/go-git/v5/plumbing/format/packfile.(*Scanner).copyObject(0xc004b9e1e0, 0xc8ecc0, 0xc016de3200, 0x0, 0x0, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/scanner.go:335 +0x35e
github.com/go-git/go-git/v5/plumbing/format/packfile.(*Scanner).NextObject(0xc004b9e1e0, 0xc8ecc0, 0xc016de3200, 0x0, 0x0, 0xe46a991753750600, 0xc004b9e230)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/scanner.go:314 +0x5a
github.com/go-git/go-git/v5/plumbing/format/packfile.(*Parser).readData(0xc00f4e50e0, 0xc8ecc0, 0xc016de3200, 0xc002c96370, 0x0, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/parser.go:414 +0x7f
github.com/go-git/go-git/v5/plumbing/format/packfile.(*Parser).get(0xc00f4e50e0, 0xc002c96370, 0xc016de3200, 0x0, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/parser.go:348 +0x38a
github.com/go-git/go-git/v5/plumbing/format/packfile.(*Parser).resolveDeltas(0xc00f4e50e0, 0x63e8598fd7ddbf07, 0xa32596928c919c4c)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/parser.go:270 +0xdb
github.com/go-git/go-git/v5/plumbing/format/packfile.(*Parser).Parse(0xc00f4e50e0, 0x0, 0x0, 0x0, 0xc0003d46e8, 0x0)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/plumbing/format/packfile/parser.go:148 +0xd3
github.com/go-git/go-git/v5/storage/filesystem/dotgit.(*PackWriter).buildIndex(0xc005c2bb90)
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/storage/filesystem/dotgit/writers.go:67 +0x1c4
created by github.com/go-git/go-git/v5/storage/filesystem/dotgit.newPackWrite
	/Users/tillsongalloway/.go/pkg/mod/github.com/go-git/go-git/v5@v5.0.0/storage/filesystem/dotgit/writers.go:53 +0x325

goroutine 891 [runnable]:
unicode.SimpleFold(0x53, 0xc000000053)
	/usr/local/opt/go/libexec/src/unicode/letter.go:339 +0x154
regexp/syntax.minFoldRune(0x73, 0xc00f790620)
	/usr/local/opt/go/libexec/src/regexp/syntax/parse.go:199 +0x4f
regexp/syntax.(*parser).newLiteral(0xc01d112fc0, 0xd500000073, 0x1)
	/usr/local/opt/go/libexec/src/regexp/syntax/parse.go:185 +0xdc
regexp/syntax.(*parser).literal(0xc01d112fc0, 0x73)
	/usr/local/opt/go/libexec/src/regexp/syntax/parse.go:210 +0x3b
regexp/syntax.Parse(0xbc26a3, 0xc1, 0xd4, 0xc90460, 0x11add60, 0x0)
	/usr/local/opt/go/libexec/src/regexp/syntax/parse.go:729 +0x2ae
regexp.compile(0xbc26a3, 0xc1, 0xd4, 0xc017a513c0, 0x1, 0x20)
	/usr/local/opt/go/libexec/src/regexp/regexp.go:168 +0x5a
regexp.Compile(...)
	/usr/local/opt/go/libexec/src/regexp/regexp.go:131
regexp.MustCompile(0xbc26a3, 0xc1, 0x0)
	/usr/local/opt/go/libexec/src/regexp/regexp.go:270 +0x4f
github.com/tillson/git-hound/internal/app.MatchKeywords(0xc017ea2d80, 0xc, 0xc, 0xc, 0xc017ea2d80)
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/keyword_scan.go:118 +0x299
github.com/tillson/git-hound/internal/app.MatchKeywords(0xc00c4ec000, 0x49fcf, 0x0, 0xc0059eb490, 0xc00b35b8b8)
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/keyword_scan.go:102 +0x231
github.com/tillson/git-hound/internal/app.GetMatchesForString(0xc00c4ec000, 0x49fcf, 0xc003f53b6a, 0xd, 0xc003f53ba6, 0xc, 0xc00677ac30, 0x21, 0xb8c4bf, 0x4, ...)
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/keyword_scan.go:292 +0x957
github.com/tillson/git-hound/internal/app.ScanAndPrintResult(0xc000392660, 0xc003f53b6a, 0xd, 0xc003f53ba6, 0xc, 0xc00677ac30, 0x21, 0xb8c4bf, 0x4, 0xc000025bd8, ...)
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/keyword_scan.go:61 +0x1ec
created by github.com/tillson/git-hound/internal/app.SearchGitHub
	/Users/tillsongalloway/.go/src/github.com/tillson/git-hound/internal/app/search.go:169 +0x8ad

```

**Comment by badaozhenjun** [comment]
without regex-file,i have the same error
```
echo mi.com | ./git-hound --dig-files --dig-commits --many-results --results-only | tee api_tokens.txt
```

**Comment by tillson** [comment]
This looks like it could be a duplicate of https://github.com/tillson/git-hound/issues/25, which was just fixed in v1.3. Feel free to re-open this issue if you're still getting errors.
